helm/templates/kafka-topic-creation-hook.yaml

```diff
@@ -37,23 +37,24 @@ spec:
             - "/bin/bash"
             - "-cex"
             - |
-            {{- range $topic := .Values.kafka.topics }}
+            {{- range $topicName, $topicConfigs := $.Values.kafka.topics }}
               /opt/kafka/bin/kafka-topics.sh --create \
               --if-not-exists \
               --zookeeper {{ $.Values.zookeeper.address | quote }} \
-              --topic {{ $topic.name | quote }} \
-            {{- range $topic.configs }}
-              --config {{ . | quote }} \
+              --topic {{ $topicName | quote }} \
+            {{- range $configName, $configValue := $topicConfigs.configs }}
+              --config "{{$configName}}={{int $configValue }}"  \
             {{- end }}
-              --replication-factor {{ $topic.replicationFactor | quote }} \
-              --partitions {{ $topic.partitions | quote }}
-            {{- if $topic.configs }}
+              --replication-factor {{ $topicConfigs.replicationFactor | quote }} \
+              --partitions {{ $topicConfigs.partitions | quote }}
+            {{- if $topicConfigs.configs }}
+            {{- $comma := false }}
               /opt/kafka/bin/kafka-configs.sh --alter \
               --zookeeper {{ $.Values.zookeeper.address | quote }} \
               --add-config \
-              {{ range $i, $c := $topic.configs }}{{ if $i }}{{ printf "," }}{{ end }}{{ $c | quote }}{{ end }} \
+              {{ range $i, $c := $topicConfigs.configs }}{{ if $comma }}{{ printf "," }}{{ end }}"{{ $i }}={{ int $c }}"{{- $comma = true -}}{{ end }} \
```

Comment:
you can't update the value of variable `comma`

Reply:
helm does not allow it.

Reply:
I actually checked it on one stackoverflow posts [(post)](https://stackoverflow.com/questions/64272789/why-doesnt-this-change-the-value-of-the-variable-in-the-range-loop-in-helm) , they say it is. 
Additionally I am getting the right rendering by executing `helm template helm/.`
Are we using older helm version ?